driver-core/src/test/unit/com/mongodb/internal/connection/TestClusterListener.java

```diff
@@ -20,17 +20,28 @@
 import com.mongodb.event.ClusterDescriptionChangedEvent;
 import com.mongodb.event.ClusterListener;
 import com.mongodb.event.ClusterOpeningEvent;
+import com.mongodb.lang.Nullable;
 
+import java.time.Duration;
 import java.util.ArrayList;
 import java.util.List;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.locks.Condition;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Predicate;
 
 import static com.mongodb.assertions.Assertions.isTrue;
 import static com.mongodb.assertions.Assertions.notNull;
+import static com.mongodb.internal.Locks.withLock;
 
-class TestClusterListener implements ClusterListener {
-    private ClusterOpeningEvent clusterOpeningEvent;
-    private ClusterClosedEvent clusterClosingEvent;
-    private final List<ClusterDescriptionChangedEvent> clusterDescriptionChangedEvents = new ArrayList<>();
+public final class TestClusterListener implements ClusterListener {
```

Comment:
The changes in `TestClusterListener` are needed for the spec tests added in this PR.